pkg/handler/scale_loop.go

```diff
@@ -34,19 +34,22 @@ func (h *ScaleHandler) HandleScaleLoop(ctx context.Context, scaledObject *kedav1
 }
 
 // handleScale contains the main logic for the ScaleHandler scaling logic.
-// It'll check each trigger active status then call scaleDeployment
+// It'll check each trigger active status then call handleScaleOnScaleObject
 func (h *ScaleHandler) handleScale(ctx context.Context, scaledObject *kedav1alpha1.ScaledObject) {
 
-	switch scaledObject.Spec.ScaleType {
-	case kedav1alpha1.ScaleTypeJob:
-		h.handleScaleJob(ctx, scaledObject)
-		break
-	default:
-		h.handleScaleDeployment(ctx, scaledObject)
-	}
+	// TODO refactor
+
+	// switch scaledObject.Spec.ScaleType {
+	// case kedav1alpha1.ScaleTypeJob:
+	// 	h.handleScaleJob(ctx, scaledObject)
+	// 	break
+	// default:
+	h.handleScaledObject(ctx, scaledObject)
+	//}
 	return
```

Comment:
I think for the scope of this PR, you ca just remove `handleScale` all together, and call `.handleScaledObject` directly in the `for/select` above. 

However, I was reviewing the code here last week and I'm thinking of some refactoring/rearranging of code to simplify things. 
Currently the `scaledobject_controller` manages the lifetime of the `scaledObject` scale loop in `ScaleHandler` with `scaleLoopContexts`. It seems a bit awkward. I'm thinking of simplifying the `ScaleHandler` contract with the controller into `go scaleHandler.RefreshScaledObject()` and `scaleHandler.ForgetScaledObject()`, _something [like this](https://github.com/ahmelsayed/keda/commit/87d9d7d5c694c0c1165f5de768bb4164c00e2853#diff-8785d7dcf69dacede8483f46f71c6066) though I was just playing around I don't like the names in the change_ 

I'll prepare a better description of the changes once I'm sure, but just wanted to say that for the scope of this PR, I think this is fine.

Reply:
Yeah, I can remove this. I was about to refactor this part of the code in the next PR, I didn't wanted to put a ton of changes into one PR.

As for the `scaleLoopContexts`, I'd be more than happy to see your proposal.